thunder/tests/test_inplace_functionalization.py

```diff
@@ -0,0 +1,124 @@
+from __future__ import annotations
+from dataclasses import dataclass
+from functools import partial
+from collections.abc import Callable
+
+import torch.testing
+
+from thunder.core import dtypes
+from thunder.core.prims import PrimIDs
+from thunder.tests.framework import ops
+from thunder.tests.opinfos import opinfos, OpInfo, make_number, SampleInput
+from thunder.tests.make_tensor import make_tensor
+from thunder.torch import _torch_to_thunder_function_map, _inplace_to_out_of_place
+
+
+# `SampleInput`s of ops with `inplace` argument do not seem to come with `inplace` arg, so give it to them.
+def sample_generator_wrapper(sample_generator):
+
+    def f(*args, **kwargs):
+        for sample in sample_generator(*args, **kwargs):
+            yield SampleInput(*(list(sample.args) + [True]), **sample.kwargs)
+
+    return f
+
+
+def inplace_masked_fill_sample_generator(op, device, dtype, requires_grad, **kwargs):
+    make = partial(make_tensor, device=device, dtype=dtype, requires_grad=requires_grad)
+    number = partial(make_number, dtype=dtype)
+
+    # pred_shape, a_shape, value
+    cases = (((2, 2, 2), (2, 2, 2), number()),)
+
+    for pred_shape, a_shape, value in cases:
+        pred, a = make(pred_shape, dtype=torch.bool, requires_grad=False), make(a_shape)
+        yield SampleInput(a, pred, value)
+
+
+_torchsymbol_to_torch: dict[Sybmol, Callable] = {v: k for k, v in _torch_to_thunder_function_map.items()}
+_functional_to_inplace: dict[Callable, Callable] = {
+    functional: inplace for inplace, (functional, index) in _inplace_to_out_of_place.items() if index == -1
+}
+_functional_to_functional_with_inplace_arg: dict[Callable, tuple[Callable, int]] = {
+    functional: (inplace, index) for inplace, (functional, index) in _inplace_to_out_of_place.items() if index >= 0
+}
+_inplace_opinfos: list[OpInfo] = []
+for op in opinfos:
+    if not (op.op in _functional_to_inplace or op.op in _functional_to_functional_with_inplace_arg):
+        continue
```

Comment:
nit: maybe in some future we would like to have a flag to test in-place ops in opinfos for out-of-place OpInfos?